### PR TITLE
feat: add enterprise_subscription_inclusion to search/all

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

Add `enterprise_subscription_inclusion` as a filterable course object field in `search/all`. Tweak previous `course_type` to be properly filterable.

## References
- https://2u-internal.atlassian.net/browse/ENT-6175
- https://github.com/openedx/course-discovery/pull/3460

### CODE REVIEW

**Review by ansabgillani — APPROVED** [review]

File: course_discovery/apps/api/v1/views/search.py [hunk 1]
```diff
@@ -191,6 +192,11 @@ class BaseAggregateSearchViewSet(FacetQueryFieldsMixin, BaseElasticsearchDocumen
         },
         'availability': {'field': 'availability.raw', 'lookups': [LOOKUP_FILTER_TERM, LOOKUP_FILTER_TERMS]},
         'content_type': {'field': 'content_type', 'lookups': [LOOKUP_FILTER_TERM, LOOKUP_FILTER_TERMS]},
+        'course_type': {'field': 'course_type', 'lookups': [LOOKUP_FILTER_TERM, LOOKUP_FILTER_TERMS]},
+        'enterprise_subscription_inclusion': {
+            'field': 'enterprise_subscription_inclusion',
+            'lookups': [LOOKUP_FILTER_TERM],
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/v1/views/search.py:198]: out of curiosity, why doesn't enterprise_subscription_inclusion have lookup in format `'lookups': [LOOKUP_FILTER_TERM, LOOKUP_FILTER_TERMS]`? What's the difference between LOOKUP_FILTER_TERM and LOOKUP_FILTER_TERMS?

**Review by DawoudSheraz — APPROVED** [review]
A question to answer, otherwise good to go.

File: course_discovery/apps/api/v1/views/search.py (hunk 1 above)
Comment (johnnagro) [reply course_discovery/apps/api/v1/views/search.py:198]: I was following the convention of other Boolean type fields, such as:

https://github.com/johnnagro/course-discovery/blob/master/course_discovery/apps/api/v1/views/search.py#L230

https://github.com/johnnagro/course-discovery/blob/master/course_discovery/apps/api/v1/views/search.py#L221

**Comment by DawoudSheraz** [comment]
@johnnagro Hi, please resolve the conflicts to make this ready for merge.

**Comment by johnnagro** [comment]
> @johnnagro Hi, please resolve the conflicts to make this ready for merge.

@DawoudSheraz resolved.

**Comment by johnnagro** [comment]
I believe this is a flapping test, can you please re-run the failed checks?

> FAILED course_discovery/apps/api/v1/tests/test_views/test_currency.py::TestCurrencyCurrencyView::test_get

**Pull request merged**